### PR TITLE
github-ci: switch to local codecov workflow

### PR DESCRIPTION
**Pull request by fliiiix**
The Codecode service has become a major disapointment,
lets try something else.
Also see this post for reference:
https://hynek.me/articles/ditch-codecov-python/

Signed-off-by: fliiiix <hi@l33t.name>

### CODE REVIEW

**Comment by fliiiix** [comment]
I don't think it makes sense to run everything twice on push and pull_request
image

**Comment by timofurrer** [comment]
Still in draft for a reason?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,10 +1,13 @@
 name: CI
 
-on: [push, pull_request]
+on: push
```
Comment (fliiiix) [inline .github/workflows/main.yml:3]: not sure if thats ok?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -40,30 +47,31 @@ jobs:
           coverage run -p -m pytest tests/integration
         env:
           PYTHONIOENCODING: UTF-8
-      - name: Report code coverage
+
+      - name: Combine coverage & fail if it's <28%.
```
Comment (fliiiix) [inline .github/workflows/main.yml:51]: we would need to upgrade that from time to time i guess

**Comment by fliiiix** [comment]
see comments

File: .github/workflows/main.yml (hunk 1 above)
Comment (timofurrer) [reply .github/workflows/main.yml:3]: If it runs for forks, too - it's fine :)

Comment (fliiiix) [reply .github/workflows/main.yml:3]: top

**Pull request merged**